### DISCUSSION TITLE
[Question]: I am unable to upload my files except for image files.

### DISCUSSION THREAD

**Discussion by sgtalh4381** [danny-avila/LibreChat#2181] [Troubleshooting] [answered]
### What is your question?

When I uploaded the image file, everything was normal. When I upload any other type of file. An "error processing file appears"

### More Details

Here are some logs from the server

2024-03-23 16:40:27 error: RAG_API_URL not set, cannot support process file upload
2024-03-23 16:40:27 error: [/files] Error processing file: RAG_API_URL not set, cannot support process file upload
2024-03-23 16:41:07 error: RAG_API_URL not set, cannot support process file upload
2024-03-23 16:41:07 error: [/files] Error processing file: RAG_API_URL not set, cannot support process file upload

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by fuegovic** [comment]
You need to setup RAG to enable support for file uploads

see: https://github.com/danny-avila/LibreChat/pull/2143
and: https://github.com/danny-avila/rag_api

If you did set it up, it looks like you're missing this in your LibreChat .env file:
- if using docker: `RAG_API_URL=http://host.docker.internal:8000`
- if not using docker: `RAG_API_URL=http://localhost:8000`

**Reply by sgtalh4381** [reply]
Thank you!

**Reply by sgtalh4381** [reply]
I am unable to use RAG

I set up the rag.yml file using the method #2143. No matter what embedded model I use. All I got was "unable to process files"

## Here are some detailed information.

### Error information for LibreChat.
```
LibreChat         | 2024-03-23 18:54:07 error: Error embedding file Request failed with status code 422
LibreChat         | 2024-03-23 18:54:08 error: [/files] Error processing file: AxiosError: Request failed with status code 422
```
### Error information for rag_api
```
rag_api-1   | INFO:     Started server process [1]
rag_api-1   | INFO:     Waiting for application startup.
rag_api-1   | INFO:     Application startup complete.
rag_api-1   | INFO:     Uvicorn running on http://0.0.0.0:8000 (Press CTRL+C to quit)
rag_api-1   | INFO:     192.168.0.1:2216 - "POST /embed HTTP/1.1" 422 Unprocessable Entity
```

**Reply by fuegovic** [reply]
I'm not sure what causing this, I have never encountered that error before. 
@danny-avila, could you please shed some light on this error when you have a moment?

**Reply by danny-avila** [reply]
Likely need to pull the latest image from rag_api


```bash
docker compose -f ./rag.yml pull
```

**Reply by danny-avila** [reply]
also use the latest commit from librechat

```bash
git checkout main
git pull
docker compose pull
docker compose up
```

**Reply by sgtalh4381** [reply]
Actually, pull the latest rag.yml file. The error I received is still "POST/embedded HTTP/1.1" 422 Unprocessable Entity.
Currently, both rag and LibreChat are running in Docker.
I haven't made any changes to the ragy.ml file.
In. env. I set ragapi to (RAG_API_URL= http://host.docker.internal:8000/)
In Nginx. I have set up a reverse proxy. Here is the configuration file for Nginx.

```
server {
    listen 443 ssl;
    server_name <domain name>;

    ssl_certificate /path/server.crt; 
    ssl_certificate_key /path/server.key; 

    location / {
        proxy_pass http://localhost:3080; 
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection 'upgrade';
        proxy_set_header Host $host;
        proxy_cache_bypass $http_upgrade;
    }
}
```

**Reply by danny-avila** [reply, marked as answer]
> Unprocessable Entity.

Can mean either you are an older image, or if you're absolutely sure you're not, your reverse proxy may be messing up the content form data

**Reply by sgtalh4381** [reply]
Everything has been updated to the latest and can run successfully! Thank you!

**Reply by danny-avila** [reply]
awesome enjoy!
